v2/compiler/container.go

```diff
@@ -111,3 +126,92 @@ func (c *workflowCompiler) addContainerDriverTemplate() string {
 	c.wf.Spec.Templates = append(c.wf.Spec.Templates, *t)
 	return name
 }
+
+func containerExecutorTemplate(container *pipelinespec.PipelineDeploymentConfig_PipelineContainerSpec) *wfapi.Template {
+	userCmdArgs := make([]string, 0, len(container.Command)+len(container.Args))
+	userCmdArgs = append(userCmdArgs, container.Command...)
+	userCmdArgs = append(userCmdArgs, container.Args...)
+	launcherCmd := []string{
+		"/kfp-launcher/launch",
+		"--execution_id", inputValue(paramExecutionID),
+		"--executor_input", inputValue(paramExecutorInput),
+		"--namespace",
+		"$(KFP_NAMESPACE)",
+		"--pod_name",
+		"$(KFP_POD_NAME)",
+		"--pod_uid",
+		"$(KFP_POD_UID)",
+		"--mlmd_server_address", // METADATA_GRPC_SERVICE_* come from metadata-grpc-configmap
+		"$(METADATA_GRPC_SERVICE_HOST)",
+		"--mlmd_server_port",
+		"$(METADATA_GRPC_SERVICE_PORT)",
+		"--", // separater before user command and args
+	}
+	mlmdConfigOptional := true
+	return &wfapi.Template{
+		Inputs: wfapi.Inputs{
+			Parameters: []wfapi.Parameter{
+				{Name: paramExecutorInput},
+				{Name: paramExecutionID},
+			},
+		},
+		Volumes: []k8score.Volume{{
+			Name: "kfp-launcher",
+			VolumeSource: k8score.VolumeSource{
+				EmptyDir: &k8score.EmptyDirVolumeSource{},
+			},
+		}},
+		InitContainers: []wfapi.UserContainer{{
+			Container: k8score.Container{
+				Name:    "kfp-launcher",
+				Image:   containerLauncherImage,
+				Command: []string{"mount_launcher.sh"},
+				VolumeMounts: []k8score.VolumeMount{{
+					Name:      "kfp-launcher",
+					MountPath: "/kfp-launcher",
+				}},
+				ImagePullPolicy: "Always",
+			},
+		}},
+		Container: &k8score.Container{
+			Command: launcherCmd,
+			Args:    userCmdArgs,
+			Image:   container.Image,
+			VolumeMounts: []k8score.VolumeMount{{
+				Name:      "kfp-launcher",
```

Comment:
Do we need to change this string like "kfp-launcher" to constants, I saw it is used in several places.

Reply:
Good point! Updated, I wrote two constants:
* volumePathKFPLauncher
* volumeNameKFPLauncher
they are both "kfp-launcher" at the moment, but they don't necessarily need to be the same